src/base/OpenMCCellAverageProblem.C

```diff
@@ -164,7 +165,9 @@ OpenMCCellAverageProblem::validParams()
   params.addParam<int>("first_iteration_particles",
                        "Number of particles to use for first iteration "
                        "when using Dufek-Gudowski relaxation");
-
+  params.addParam<bool>("use_displaced_mesh",
```

Comment:
@nuclearkevin does this need to be modified at all to be consistent with what you did with `fixed_mesh` being auto-detected? 

@meltawila can you please also remind me, was there a use case where we wanted to disable the displaced mesh updates in OpenMC, even if the mesh is actually moving? Or was this user option only existing because we didn't know how to detect it automatically?

Reply:
This is the logic used in MOOSE solid mechanics, to be able to set `use_displaced_mesh = false` for certain post processors/user objects only for example

Reply:
@aprilnovak At the moment this doesn't need to be modified since the changes made to auto-detect `fixed_mesh` are coming in the AMR PR (#964). If this gets merged first, I can handle the necessary changes in #964. If the AMR PR gets merged ahead of this one, a single line of code will need to be added to this PR to get auto-detection working:
https://github.com/neams-th-coe/cardinal/pull/964/files#diff-7321f62ca7502f529d9917464fed4643c5e57c97c27d295f78db837d1a35a366R219

Reply:
For reference @meltawila this is what's necessary to automatically detect if the mesh is moving and setup the `OpenMCCellAverageProblem` accordingly. This include is required:
```
#include "CreateDisplacedProblemAction.h"
```
and this needs to get added to the constructor of `OpenMCCellAverageProblem`:
```
// Check to see if a displaced problem is being initialized.
const auto & dis_actions = getMooseApp().actionWarehouse().getActions<CreateDisplacedProblemAction>();
for (const auto & act : dis_actions)
{
  auto has_displaced = act->isParamValid("displacements") && act->getParam<bool>("use_displaced_mesh");
  _need_to_reinit_coupling |= (has_displaced && _use_displaced_mesh);
}
```